### PR TITLE
add `nospecialize` to `invokelatest`

### PR DESCRIPTION
**Pull request by JeffBezanson**
Also change an `eval` to `invokelatest` in LineEdit --- I'm hoping somebody can confirm that this is correct.

### CODE REVIEW

**Comment by StefanKarpinski** [comment]
Would that someone be @vtjnash?

**Comment by JeffBezanson** [comment]
Or somebody familiar with that LineEdit code.

**Comment by Keno** [comment]
Yes, I think it's correct.

File: base/essentials.jl [hunk 1]
```diff
@@ -662,7 +662,7 @@ call obsolete versions of a function `f`.
 (The drawback is that `invokelatest` is somewhat slower than calling
 `f` directly, and the type of the result cannot be inferred by the compiler.)
 """
-function invokelatest(f, args...; kwargs...)
+function invokelatest(@nospecialize(f), @nospecialize args...; kwargs...)
     # We use a closure (`inner`) to handle kwargs.
     inner() = f(args...; kwargs...)
     Core._apply_latest(inner)
```
Comment (Keno) [inline base/essentials.jl]: While we're in the process of optimizing this, might it make sense to check for empty kwargs here and if so use `Core._apply_latest(f, args)`?

**Review by Keno — APPROVED** [review]

**Comment by Keno** [comment]
I've restarted AppVeyor. They both failed, but the log was incomplete, so I wasn't able to see why. Perhaps an AppVeyor fluke.

**Comment by Keno** [comment]
AppVeyor on this PR seems to take longer than on master on average. I'll merge, but let's keep a look on that and investigate if it turns out to persist.

**Comment by Sacha0** [comment]
> AppVeyor on this PR seems to take longer than on master on average. I'll merge, but let's keep a look on that and investigate if it turns out to persist.

I have seen AV run in excess of three hours on other pull requests lately, if that's the order of longer you mean. Best!

**Pull request merged**